src/GovUk.Education.ExploreEducationStatistics.Admin/Services/Interfaces/IPublicationService.cs

```diff
@@ -28,8 +28,16 @@ public interface IPublicationService
 
         Task<Either<ActionResult, PublicationViewModel>> GetPublication(Guid publicationId);
 
-        Task<Either<ActionResult, List<ReleaseViewModel>>> ListActiveReleases(Guid publicationId);
+        Task<Either<ActionResult, PaginatedListViewModel<ReleaseListItemViewModel>>> ListActiveReleasesPaginated(
```

Comment:
I wonder if we bother keeping both `ListActiveReleasesPaginated` and `ListActiveReleases`, or instead just have one method with the pagination support and just let code that needs all of them call it with "unlimited" page size?

Reply:
So in the case of my other comment about bundling `page` and `pageSize` into a `PaginationParams` record, we could have a factory method on `PaginationParams` called something like `NoPagination`.  Thus a caller would say:

`var allReleases = publicationService.ListActiveReleases(publicationId, PaginationParams.NoPagination());`

or:

`var paginatedReleases = publicationService.ListActiveReleases(publicationId, paginationParams);`

depending on the use case?

Reply:
Or better, a singleton called `NoPagination` within `PaginationParams` that represents no pagination. Thus callers say:

`var allReleases = publicationService.ListActiveReleases(publicationId, PaginationParams.NoPagination);`

Reply:
OOOOR!

Given that all pagination parameters could now be in a single nullable parameter:

```
Task<Either<ActionResult, PaginatedListViewModel<ReleaseListItemViewModel>>> ListActiveReleases(            
            Guid publicationId,
            PaginationParams pagination,
            bool? live = null);
```

a caller would say:

`var allReleases = publicationService.ListActiveReleases(publicationId);`

or:

`var paginatedReleases = publicationService.ListActiveReleases(publicationId, paginationParams);`

Reply:
So my thought behind keeping them separate is to try to stop pagination concerns infecting lots of the codebase if it becomes more widespread. Right now, since `ListActiveReleases` returns all results, it has been simple to separate out pagination into its own thing. I was thinking that if in the future we wanted to reduce the load on the DB and only return results from the DB for the requested page, then maybe `ListActiveReleases` could return a `Queryable` which could then be used by a generic pagination method that gets the correct page for any given queryable.

This has two advantages in my eyes - pagination and the stuff being paginated can be unit tested independently (we would test pagination once rather than repeatedly for each individual endpoint that has pagination, and you don't need to think about pagination when testing methods like `ListActiveReleases`) and someone writing a new feature gets the pagination at minimal cost (they just need to ensure their new method provides a `Queryable` that a generic pagination method could consume).

What got me thinking along these lines is that there are several things in the codebase that probably have infected the codebase and are gotchas if you forget to account for them when adding a feature - and I suspect it didn't have to be this way. 

All that said, I doubt pagination will hold such sway so maybe it's not such a big concern in this case, but on the other hand if you entangle stuff I think you often get unforeseen consequences.

So my preference would be to keep the two different methods, but happy to do whatever you think is best. Happy to add paginationParams - although not sure I'm going to get the time today!